### ISSUE TITLE
There are errors in test codes that expect the subscription to emit onError.

### ISSUE THREAD

**Issue by naxxster** [bjornbytes/RxLua#37]
There are errors in test codes that expect the subscribe function to fail.
The tests fail not because of emitting onError, but because of failing to invoke the subscribe function.

Here is an example from tests/all.lua
```lua
  it('passes through errors', function()
    expect(Rx.Observable.throw():all().subscribe).to.fail()
  end)
```
The error description is "attempt to index local 'self' (a nil value)".

I think this should be
```lua
  it('passes through errors', function()
    expect(function () Rx.Observable.throw():all():subscribe() end).to.fail()
  end)
```

**Comment by naxxster** [comment]
I think it's better to have an assertion like
```lua
expect(observable).to.produce.error(...)
```
This distinguishes other errors from observable's onError.

**Comment by bjornbytes** [comment]
Fixed in #39
